src/hotspot/share/cds/aotClassInitializer.cpp

```diff
@@ -28,32 +28,139 @@
 #include "cds/cdsConfig.hpp"
 #include "oops/instanceKlass.inline.hpp"
 
+class AOTClassInitializer::AllowedSpec {
```

Comment:
Suggestion:

// An AllowedSpec is used to match a class name Symbol against either the full name text or a leading prefix.